script.js

```diff
@@ -150,7 +213,157 @@ function calculateUpdatedTotalPriceOrder(basket) {
+
+    if (document.querySelector('.total-price-order').innerHTML === information) {
+        document.querySelector('#delete-all').style.display = 'none';
+    } else {
+        document.querySelector('#delete-all').style.display = 'block';
+    }
+}
+
+/**
+ * Clearing basket
+ */
+function addEventListenerForDeleteAllIteamsButton() {
+
+    document.querySelector('#delete-all').addEventListener('click', function () {
+        basket = [];
+        renderBasket();
+    });
+}
+
+/**
+ * Check if there is already a value in local storage and load basket
+ */
+function loadLocalStorage() {
+
+    if (!JSON.parse(localStorage.getItem("basket"))) {
+        localStorage.setItem("basket", JSON.stringify(basket));
+    } else {
+        basket = JSON.parse(localStorage.getItem("basket"))
+    }
+}
+
+function addSortListeners() {
+
+    document.querySelector('#sort-ascending-title-button').addEventListener('click', () => sortAscendingByTitle(true));
+    document.querySelector('#sort-descending-title-button').addEventListener('click', sortDescendingByTitle);
+    document.querySelector('#sort-ascending-price-button').addEventListener('click', sortAscendingByPrice);
+    document.querySelector('#sort-descending-price-button').addEventListener('click', sortDescendingByPrice);
+}
+
+/**
+ * Sort pizzas in ascending order by price.
+ */
+function sortAscendingByPrice() {
+
+    pizzaList.sort(function (a, b) {
+        return a.price - b.price
+    });
+    showAllPizzas();
+}
+
+/**
+ * Sort pizzas in descending order by price.
+ */
+function sortDescendingByPrice() {
+
+    pizzaList.sort(function (a, b) {
+        return b.price - a.price
+    });
+    showAllPizzas();
+}
+
+/**
+ * Sort pizzas in ascending order by title.
+ */
+function sortAscendingByTitle(show) {
+
+    pizzaList.sort(function (a, b) {
+        if (a.title < b.title) {
+            return -1;
+        }
+        if (b.title < a.title) {
+            return 1;
+        }
+        return 0;
+    });
+
+    if (show === true) {
+        showAllPizzas();
+    }
+}
+
+/**
+ * Sort pizzas in descending order by title.
+ */
+function sortDescendingByTitle() {
+
+    pizzaList.sort(function (a, b) {
+        if (a.title > b.title) {
+            return -1;
+        }
+        if (b.title > a.title) {
+            return 1;
+        }
+        return 0;
+    });
+    showAllPizzas();
+}
+
+/**
+ * Filtering by ingredient.
+ * @param {*} query string from search bar.
+ * @param {*} event to detect backspace.
+ */
+function filterOnPizzas(query, event) {
+    "use strict";
+    let temp = [];
+
+    if (event.inputType === "deleteContentBackward") {
+        pizzaList = globalTempPizzaList;//renderPizzasT(); 
+    }
+
+    if (pizzaList.length === 0) {
+        pizzaList = globalTempPizzaList; //renderPizzasT();
+    }
+
+    if (query !== "") {
+        let next = [];
+        let result = [];
+        query.split(',').forEach(searchedIngredient => {
```

Comment:
ta funkcja od filtrowania wyglada na bardzo przekombinowana, lepiej pobrac wyszukiwania

```
const queryItems = query.split(',').map(item => item.trim())
```

a potem zrobic jedynie filter
```
globalTempPizzaList.filter(pizza => pizza.ingredients.some(ingredient => queryItems.some(queryItem => ingredient.includes(queryItem))))
```

mozesz tez to podzielic na mniejsze czesci jezeli jest bardziej czytelne dla Ciebie, ale postaraj sie przepisac ta funkcje